internal/chain/beacon/node.go

```diff
@@ -353,33 +342,23 @@ func (h *Handler) IsStopped() bool {
 	return h.stopped
 }
 
-func (h *Handler) Reset(ctx context.Context) {
-	_, span := metrics.NewSpan(ctx, "h.Reset")
-	defer span.End()
-
-	h.Lock()
-	defer h.Unlock()
-
-	h.stopped = false
-	h.started = false
-	h.running = false
-	h.serving = false
-}
-
 // run will wait until it is supposed to start
 //
-//nolint:funlen // This is a long function
+//nolint:funlen // this is a big function
 func (h *Handler) run(startTime int64) {
+	// we cannot re-start a stopped handler
+	if h.IsStopped() {
+		return
+	}
+	h.Lock()
+	h.running = true
```

Comment:
all these notches feel a bit weird - maybe not important for this PR, but it might make more sense to turn this into a state machine with `iota` and have `{ Running, Serving, Stopped }` or something